spec/src/main/asciidoc/mpconfig.asciidoc

```diff
@@ -101,3 +101,35 @@ org.eclipse.microprofile.example.MyBean.createExecutor.1.maxAsync=10
 ----
 
 Again, it would be wrong to specify `org.eclipse.microprofile.example.MyBean.exec6.maxAsync`, because the container does not produce a new instance for the `exec6` injection point. The container produces the new instance for the `exec` injection point, and matches that same instance for injection into `exec6` per the `@NamedInstance("executor6")` qualifier.
+
+=== Overriding Array Properties in MicroProfile Config
+
+MicroProfile Config can be used to override array type properties (`propagated`, `cleared`, and `unchanged`) on injection points for which the container creates `ManagedExecutor` or `ThreadContext` instances. The following rules apply for config property values:
+
+- The value can be a single array element, multiple elements (delimited by `,`), or empty.
+- Array elements can be any value returned by a ``ThreadContextProvider``'s `getThreadContextType()` method.
+- Array elements can be any thread context type constant value from ThreadContext (such as `Security`, `Application`, or `Remaining`).
+- The usual rules from the MicroProfile Config specification apply, such as escaping special characters.
+
+For example, we can start with the following injection point, which has the default configuration for `ThreadContext`. This means clearing the `Transaction` context, leaving no context type unchanged, and propagating all `Remaining` context types,
+
+[source, java]
+----
+package org.eclipse.microprofile.example;
+...
+public class MyBean {
+    @Inject
+    ThreadContext appAndCDIContextPropagator;
```

Comment:
I would choose more neutral field name(`configurableContextPropagator`?), this one can be confusing as from the code itself it looks like it has default settings whereas the actual behaviour depends on the settings described below.

Reply:
Yes, I can update that name to make it more generic